File: kotlin/151.kt
class Solution {
    fun reverseWords(s: String): String {
        return s
            .split(" ")
            .filter { it != "" }
            .reversed()
            .joinToString(" ")
    }
}

fun main() {
    val solution = Solution()

    println(solution.reverseWords("word1 word2 word3  word4"))
    println(solution.reverseWords("   word1   word2  word3  word4       "))
    println(solution.reverseWords(""))
    println(solution.reverseWords("word1"))
}

File: javascript/151.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    return s
        .split(" ")
        .filter(input => input != "")
        .reverse()
        .toString()
        .replaceAll(",", " ")
};

const input1 = "the sky is blue"
const input2 = "  hello world  "
const input3 = "a good   example"

console.log( reverseWords(input1) )
console.log( reverseWords(input2) )
console.log( reverseWords(input3) )